tests/test_operators.py

```diff
@@ -1055,33 +1055,55 @@ def test_is_not_unique_relationship(self):
         )
 
     def test_empty_within_except_last_row(self):
-        df = pandas.DataFrame.from_dict(
+        valid_df = pandas.DataFrame.from_dict(
+            {
+                "USUBJID": [789, 789, 789, 789, 790, 790, 790, 790, ],
+                "SESEQ": [1, 2, 3, 4, 5, 6, 7, 8, ],
+                "SEENDTC": ["2006-06-03T10:32", "2006-06-10T09:47", "2006-06-17", "", "2006-06-03T10:14", "2006-06-10T10:32", "2006-06-17", "2006-06-17"],
+                "SESTDTC": ["2006-06-01", "2006-06-03T10:32", "2006-06-10T09:47", "2006-06-17", "2006-06-01", "2006-06-03T10:14", "2006-06-10T10:32", "2006-06-17"],
+            }
+        )
+        invalid_df = pandas.DataFrame.from_dict(
             {
-                "USUBJID": [1, 1, 1, 2, 2, 2],
-                "valid": ["2020-10-10", "2020-10-10", "2020-10-10", "2021", "2021", "2021", ],
-                "invalid": ["2020-10-10", None, None, "2020", "2020", None, ],
+                "USUBJID": [789, 789, 789, 789, 790, 790, 790, 790, ],
+                "SESEQ": [1, 2, 3, 4, 5, 6, 7, 8, ],
+                "SEENDTC": ["", "2006-06-10T09:47", "2006-06-17", "", "2006-06-03T10:14", "2006-06-10T10:32", "2006-06-17", "2006-06-17"],
+                "SESTDTC": ["2006-06-01", "2006-06-03T10:32", "2006-06-10T09:47", "2006-06-17", "2006-06-01", "2006-06-03T10:14", "2006-06-10T10:32", "2006-06-17"],
             }
         )
-        self.assertFalse(
-            DataframeType({"value": df}).empty_within_except_last_row({"target": "valid", "comparator": "USUBJID"})
```

Comment:
The ordering property should be optional, so we should still retain the old tests that don't have the ordering property

Reply:
OK done